plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -34,17 +47,60 @@
   public static final String OAUTH_CLIENT_ID = "@oauth.client.id@";
   public static final String OAUTH_CLIENT_SECRET = "@oauth.client.secret@";
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email",
+      "https://www.googleapis.com/auth/cloud-platform"
   ));
 
   /**
    * Returns the credential of the active user. If there is no active user, returns {@code null}.
    */
   // Should probably be synchronized properly.
   // TODO(chanseok): consider returning a String JSON (i.e., hide Credential)
-  public Credential getActiveCredential() {
-    return null;
+  public Credential getActiveCredential() throws IOException {
+    GoogleTokenResponse authResponse = logIn();
+    return createCredential(authResponse);
+  }
+
+  private GoogleTokenResponse logIn() throws IOException {
+    // 1. Open a browser that takes care of the login.
+    GoogleAuthorizationCodeRequestUrl requestUrl = new GoogleAuthorizationCodeRequestUrl(
+        OAUTH_CLIENT_ID, GoogleOAuthConstants.OOB_REDIRECT_URI, OAUTH_SCOPES);
+
+    try {
+      IWorkbenchBrowserSupport browserSupport = PlatformUI.getWorkbench().getBrowserSupport();
```

Comment:
Thanks. Copying verification codes should certainly go away. I'll also follow the `LoginBrowser` flow, which I think I need setup a local web server to listen for a verification code. This will require a fairly large amount of change, so I put another TODO item in #342.